### PR TITLE
Aet ui schema viewer

### PR DESCRIPTION
**Pull request by lordmallam**
#### Related JIRA Ticket [AET-264](https://jira.ehealthafrica.org/browse/AET-264)

Bug fix: Avro schema viewer crashes when unions are applied.  Applies to `Entity Type Viewer` as well

### CODE REVIEW

**Review by obdulia-losantos — CHANGES_REQUESTED** [review]
Could you include these cases in the tests?

File: aether-ui/aether/ui/assets/apps/pipeline/redux.jsx [hunk 1]
```diff
@@ -100,11 +100,12 @@ const parsePipeline = (pipeline) => {
...
     // find out the number assigned to the linked Entity Type
-    const entityType = rule.destination.split('.')[0]
+    const ruleSplit = rule.destination.split('.')
+    const entityType = ruleSplit[0] === '$' ? ruleSplit[1] : ruleSplit[0]
     const color = entityColors[entityType] || 0
 
-    highlightSource[rule.source] = color
-    highlightDestination.push(rule.destination)
+    highlightSource[rule.source.replace('$.', '')] = color
+    highlightDestination.push(rule.destination.replace('$.', ''))
```
Comment (obdulia-losantos) [inline aether-ui/aether/ui/assets/apps/pipeline/redux.jsx]: The destination fields don't have the `$` prefix, do they? :thinking:

**Comment by obdulia-losantos** [comment]
This is the result of using the AVRO schema generated from an XForm... 

image

This is the expected tree (I removed some of the unions), the union types can contain really complicated nested types.

image

Sorry, I never said this was going to be easy :disappointed_relieved:

**Comment by obdulia-losantos** [comment]
Is this on purpose? I mean the ~index~, ~value~,...

image

**Comment by obdulia-losantos** [comment]
I'm going to suggest to transform the mock files from `jsx` to `json` this will let us use them just copying/pasting them :grimacing:

**Comment by obdulia-losantos** [comment]
An edge case, union with two different types not only null, but do not try to implement this case ;-)

```json
{
  "name": "PlayingWithCoords",
  "type": "record",
  "fields": [
    {
      "name": "location",
      "type": [
        "null",
        {
          "name": "location_geopoint",
          "type": "record",
          "fields": [
            {
              "name": "coordinates",
              "type": {
                "type": "array",
                "items": "float"
              }
            },
            {
              "name": "altitude",
              "type": [
                "null",
                "float"
              ]
            },
            {
              "name": "accuracy",
              "type": [
                "null",
                "float"
              ]
            },
            {
              "name": "type",
              "type": [
                "null",
                "string"
              ]
            }
          ]
        },
        {
          "name": "location_lat_lng",
          "type": "record",
          "fields": [
            {
              "name": "latitude",
              "type": [
                "null",
                "float"
              ]
            },
            {
              "name": "longitude",
              "type": [
                "null",
                "float"
              ]
            },
            {
              "name": "altitude",
              "type": [
                "null",
                "float"
              ]
            },
            {
              "name": "accuracy",
              "type": [
                "null",
                "float"
              ]
            }
          ]
        }
      ]
    }
  ]
}
```

File: aether-ui/aether/ui/assets/tests/mock/schema_entityTypes.mock.jsx [hunk 2]
```diff
@@ -13,11 +13,11 @@ const entityTypes = [
       },
       {
         'name': 'lastName',
-        'type': 'string'
+        'type': ['null', 'string']
       },
       {
         'name': 'age',
-        'type': 'int'
+        'type': ["int", {"name": "data-type", "type": "string"}]
```
Comment (obdulia-losantos) [inline aether-ui/aether/ui/assets/tests/mock/schema_entityTypes.mock.jsx]: `standard` should have complained here :scream:

Comment (obdulia-losantos) [reply aether-ui/aether/ui/assets/tests/mock/schema_entityTypes.mock.jsx]: AS I mentioned before, it's better to transform this into JSON and use double quotes :wink:

**Comment by lordmallam** [comment]
Yes the `line-strike` ~index~, ~value~ is on purpose to indicate array items as they are not directly referenced as other properties. The style will be changed to properly represent array items. Any suggestions on how this can be indicated?

File: aether-ui/aether/ui/assets/tests/mock/schema_entityTypes.mock.jsx (hunk 2 above)
Comment (lordmallam) [reply aether-ui/aether/ui/assets/tests/mock/schema_entityTypes.mock.jsx]: 🤔

**Comment by obdulia-losantos** [comment]
:flushed: 

SCSS lint warnings

```
./aether/ui/assets/css/_section-input.scss
  63:22  warning  Trailing semicolons required  trailing-semicolon

./aether/ui/assets/css/_section-output.scss
  16:1  warning  Space expected between blocks  empty-line-between-blocks
```

React warnings

```
    Warning: Each child in an array or iterator should have a unique "key" prop.
    
    Check the render method of `AvroSchemaViewer`. See https://fb.me/react-warning-keys for more information.
        in li (created by AvroSchemaViewer)
        in AvroSchemaViewer (created by WrapperComponent)
        in WrapperComponent
```

**Review by obdulia-losantos — APPROVED** [review]

**Pull request merged**